src/main/java/com/sngular/kloadgen/sampler/SamplerUtil.java

```diff
@@ -126,12 +127,14 @@ public static Properties setupCommonProperties(final JavaSamplerContext context)
     props.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, context.getParameter(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG));
     props.put(ProducerKeysHelper.SASL_MECHANISM, context.getParameter(ProducerKeysHelper.SASL_MECHANISM));
 
-    String schemaRegistryNameValue = context.getJMeterVariables().get(SchemaRegistryKeyHelper.SCHEMA_REGISTRY_NAME);
-    String enableSchemaRegistrationValue = context.getParameter(SchemaRegistryKeyHelper.ENABLE_AUTO_SCHEMA_REGISTRATION_CONFIG);
+    final String schemaRegistryNameValue = context.getJMeterVariables().get(SchemaRegistryKeyHelper.SCHEMA_REGISTRY_NAME);
+    final String enableSchemaRegistrationValue = context.getParameter(SchemaRegistryKeyHelper.ENABLE_AUTO_SCHEMA_REGISTRATION_CONFIG);
     if (schemaRegistryNameValue.equalsIgnoreCase(SchemaRegistryKeyHelper.SCHEMA_REGISTRY_APICURIO)) {
```

Comment:
May be this code could need a refactor:
First, I would change the order in the condition to avoid problem with null values:
`SchemaRegistryKeyHelper.SCHEMA_REGISTRY_APICURIO(schemaRegistryNameValue.equalsIgnoreCase)`
Second, and may be it's not a good idea (I even don't like it), something like:
```
private void giveMeAProperName(final Object props, String firstKey, String firstValue, String secondKey, String secondValue){
props.put(firstKey, firstValue);
      props.put(secondKey, context.getJMeterVariables().get(secondValue));
}
...
if(SchemaRegistryKeyHelper.SCHEMA_REGISTRY_APICURIO(schemaRegistryNameValue.equalsIgnoreCase)){
  giveMeAProperName(props, SerdeConfig.AUTO_REGISTER_ARTIFACT, enableSchemaRegistrationValue, SchemaResolverConfig.REGISTRY_URL));
} else {
giveMeAProperName(props, SchemaRegistryKeyHelper.ENABLE_AUTO_SCHEMA_REGISTRATION_CONFIG, enableSchemaRegistrationValue, SchemaRegistryKeyHelper.SCHEMA_REGISTRY_URL));
}
```

Reply:
I reverse the order of the condition